libs/libc/audio/libsrc/Make.defs

```diff
@@ -0,0 +1,49 @@
+############################################################################
+# audio/libsrc/Make.defs
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+ifeq ($(CONFIG_AUDIO_SRC),y)
+
+libsamplerate:
+	$(Q) wget https://codeload.github.com/libsndfile/libsamplerate/zip/master -O libsamplerate.zip
+	$(Q) unzip -o libsamplerate.zip
+	$(Q) mv libsamplerate-master libsamplerate
+	$(Q) patch -Np1 --verbose -d libsamplerate <audio/libsrc/0001-samplerate-define-package-and-version-for-NuttX-OS.patch
+	$(Q) patch -Np1 --verbose -d libsamplerate <audio/libsrc/0002-add-audio-quality-options.patch
+	$(Q) cp -rf libsamplerate/src/samplerate.h $(TOPDIR)$(DELIM)include$(DELIM)nuttx$(DELIM)audio$(DELIM)
+
+depend: libsamplerate
```

Comment:
depend::?

Reply:
from my understanding should work with : instead of ::

Reply:
No, if you want to depend appear more than once(in libc/Makefile and audio/libsrc/Make.defs), you must use :: in both location which mean that clean, depend and context need change to :: in both makefile.
BTW, we should use context here instead of depend.

Reply:
+1 for context.
In the same note, the clean should happen in "clean_context" and "distclean" not in "clean".
However, libc isn't part of the clean_context directories.  To include it we need to add libc in Directories.mk, use a double colon target in libc/Makefile for clean_context and extend it in libc/audio/Make.defs.

Reply:
@xiaoxiang781216 thanks for the explanation.
I updated the patch with the :: and tested the functionality

Reply:
But I don't see there are any tough reason to not do the right thing initially.